File: ts_code/MoveZeroes.ts
class MoveZeroes {
    public static moveZeroes(nums: number[]): void {
        let lastNonZeroFoundAt = 0; // ③

        // ⑤
        for (let i = 0; i < nums.length; i++) {
            if (nums[i] !== 0) { // ⑥
                const temp = nums[lastNonZeroFoundAt]; // ⑦
                nums[lastNonZeroFoundAt] = nums[i];    // ⑧
                nums[i] = temp;                        // ⑨
                lastNonZeroFoundAt++;                  // ⑩
            }
        }
    }
}

// Ejemplo de uso:
const nums_1: number[] = [0, 1, 0, 3, 12];
MoveZeroes.moveZeroes(nums_1);
console.log("Resultado:", nums_1); // Resultado esperado: [1, 3, 12, 0, 0]

File: java_code/MoveZeroes.java
package java_code;

public class MoveZeroes {
    public static void moveZeroes(int[] nums) {
        // 2
        int lastNonZeroFoundAt = 0; // 3
        // 4
        for (int i = 0; i < nums.length; i++) { // 5
            if (nums[i] != 0) { // 6
                int temp = nums[lastNonZeroFoundAt]; // 7
                nums[lastNonZeroFoundAt] = nums[i]; // 8
                nums[i] = temp; // 9
                lastNonZeroFoundAt++; // 10
            } // 11
        } // 12
    }

    public static void main(String[] args) {
        int[] nums1 = { 0, 1, 0, 3, 12 };
        moveZeroes(nums1);
    }
}
